tests/plugins/test_koji_promote.py

```diff
@@ -267,13 +266,19 @@ def mock_environment(tmpdir, session=None, name=None,
         sync_result = {}
         for image in workflow.tag_conf.images:
             tag = image.to_str(registry=False)
-            digests = ManifestDigest(v1=fake_digest(image, 1),
-                                     v2=fake_digest(image, 2))
-            docker_reg.digests[tag] = digests
+            man_fake = ManifestDigest(v1=fake_digest(image, 1),
+                                      v2=fake_digest(image, 2))
+
+            if digests:
+                man_digests = ManifestDigest(**digests)
+            else:
+                man_digests = man_fake
+
+            docker_reg.digests[tag] = man_digests
             if pulp_registries:
-                sync_result[digests.v1] = {}
+                sync_result[man_fake.v1] = {}
```

Comment:
Not sure why this isn't `sync_result[man_digests.v1] = {}`. Is it just that we don't test with pulp_registries as well as digests?

Reply:
well we want to feed pulp registry always with digests, but man_digests are only meant for docker registries, so it might be that man_digests doesn't have that type of digest, so I have to use the fake one

Reply:
Ah, the distiction is that 'digests' is for the digests available in the docker registry, but 'sync_result' carries the digests available in Pulp after sync.

Can we make the naming reflect that more clearly? Maybe `registry_digests` instead of `digests` for the mock_environment parameter name?

Then I think we always want the v2 schema 2 digest to be identical in the registry and in Pulp. Only the v2 schema 1 digest might differ.

So I think *this* line is fine, just the variable names might be improved.

Reply:
ok will rename that to 'registry_digests'